math/src/elliptic_curve/short_weierstrass/curves/bls12_381/compression.rs

```diff
@@ -13,12 +14,14 @@ use std::ops::Neg;
 
 pub type G1Point = ShortWeierstrassProjectivePoint<BLS12381Curve>;
 pub type BLS12381FieldElement = FieldElement<BLS12381PrimeField>;
-const MODULUS: U256 =
+pub const SUBGROUP_ORDER: U256 =
     U256::from_hex_unchecked("73eda753299d7d483339d80809a1d80553bda402fffe5bfeffffffff00000001");
 
-pub fn check_point_is_in_subgroup(point: &G1Point) -> bool {
-    let inf = G1Point::neutral_element();
-    let aux_point = point.operate_with_self(MODULUS);
+pub fn check_point_is_in_subgroup<SW: IsShortWeierstrass>(
+    point: &ShortWeierstrassProjectivePoint<SW>,
+) -> bool {
+    let inf = ShortWeierstrassProjectivePoint::<SW>::neutral_element();
+    let aux_point = point.operate_with_self(SUBGROUP_ORDER);
     inf == aux_point
```

Comment:
Here you can use a is a specific optimized method to check whether a point is the neutral element ([here](https://github.com/lambdaclass/lambdaworks/blob/21cc017bba911e0ada0822b291a180c393616c3b/math/src/elliptic_curve/short_weierstrass/point.rs#L83C18-L86)). In the case of Short Weierstrass, a point on the curve is the neutral element if and only if its Z coordinate is zero. So there's no need to compare all three coordinates.

Reply:
Thank you!